test/test_lifetime.jl

```diff
@@ -1,80 +1,285 @@
-                            @test value.(m[:cap_inst][source, t]) ==
-                                  EMB.capacity(source, t_inv) +
-                                  value.(m[:cap_add][source, t_inv])
-                            break
-                        end
-                    end
-                end
-            end
-        end
-        @testset "Cost comparisons - $(sp_dur) years" begin
-            if sp_dur > lifetime
-                @test floor(resulting_obj["$(sp_dur) years"][3]) ==
-                      floor(resulting_obj["$(sp_dur) years"][4])
-            elseif sp_dur == lifetime
-                @test floor(resulting_obj["$(sp_dur) years"][3]) ==
-                      floor(resulting_obj["$(sp_dur) years"][4])
-                @test floor(resulting_obj["$(sp_dur) years"][2]) ==
-                      floor(resulting_obj["$(sp_dur) years"][3])
-            end
-            if sp_dur * 4 < lifetime #4 corresponds to the len of T, i.e. the number of strategic periods.
-                @test floor(resulting_obj["$(sp_dur) years"][2]) ≈
-                      floor(resulting_obj["$(sp_dur) years"][4])
-            end
-        end
+@testset "UnlimitedLife" begin
+    # Creation and solving of the model
+    demand = StrategicProfile([10,30,30,20])
+    m, para = simple_model(;demand)
+
+    # Extraction of required data
+    n = para[:node]
+    𝒯 = para[:T]
+    𝒯ᴵⁿᵛ = strat_periods(𝒯)
+    inv_data = para[:inv_data]
+    capex = StrategicProfile([1,1,1,0])*1e4
+
+    # Tests of the lifetime calculation
+    # - set_capacity_cost(m, element, inv_data, prefix, 𝒯ᴵⁿᵛ, disc_rate, ::UnlimitedLife)
+    @testset "Lifetime calculations" begin
+        # Test that `:cap_rem` is fixed to 0
+        @test sum(is_fixed(m[:cap_rem][n, t_inv]) for t_inv ∈ 𝒯ᴵⁿᵛ) ==
+            length(𝒯ᴵⁿᵛ)
+    end
+
+    # Test that the CAPEX is correctly calculated
+    # - set_capex_value(m, element, inv_data, prefix, 𝒯ᴵⁿᵛ, ::Investment)
+    @testset "Capex calculation" begin
+        @test sum(
+            value.(m[:cap_capex])[n, t_inv] ==
+                value.(m[:cap_add])[n, t_inv] * EMI.capex(inv_data, t_inv) for
+            t_inv ∈ 𝒯ᴵⁿᵛ) == length(𝒯ᴵⁿᵛ)
+        @test sum(
+            value.(m[:cap_capex])[n, t_inv] == capex[t_inv] for
+            t_inv ∈ 𝒯ᴵⁿᵛ) == length(𝒯ᴵⁿᵛ)
+    end
+end
+
+@testset "StudyLife" begin
+    # Creation and solving of the model
+    inv_data = NoStartInvData(
+        FixedProfile(1000),
+        FixedProfile(40),
+        ContinuousInvestment(FixedProfile(0), FixedProfile(15)),
+        StudyLife(FixedProfile(20))
+    )
+    demand = StrategicProfile([10,10,30,35])
+    m, para = simple_model(;inv_data,demand)
+
+    # Extraction of required data
+    n = para[:node]
+    𝒯 = para[:T]
+    𝒯ᴵⁿᵛ = strat_periods(𝒯)
+    inv_data = para[:inv_data]
+    disc_rates = [objective_weight(t_inv, Discounter(para[:disc_rate], 𝒯)) for t_inv ∈ 𝒯ᴵⁿᵛ]
+    capex = StrategicProfile([
```

Comment:
Perhaps explain the numbers and calculation of capex, not sure I follow why the updates are different, e.g. 15 is not updated.